File: gherkin/java/src/main/java/gherkin/pickles/PickleStep.java
package gherkin.pickles;

import java.util.List;

import static java.util.Collections.unmodifiableList;

public class PickleStep {
    private final String text;
    private final List<Argument> arguments;
    private final List<PickleLocation> locations;

    public PickleStep(String text, List<Argument> arguments, List<PickleLocation> locations) {
        this.text = text;
        this.arguments = unmodifiableList(arguments);
        this.locations = unmodifiableList(locations);
    }

    public String getText() {
        return text;
    }

    public List<PickleLocation> getLocations() {
        return locations;
    }

    public List<Argument> getArgument() {
        return arguments;
    }
}

File: gherkin/dotnet/Gherkin/Pickles/PickleStep.cs
using System.Collections.Generic;
using System.Runtime.Serialization;
using Gherkin.Stream;

namespace Gherkin.Pickles
{
    public class PickleStep
    {
        [DataMember(Name = "id")]
        public string Id { get; set; }

        [DataMember(Name = "text")]
        public string Text { get; set; }

        [DataMember(Name = "astNodeIds")]
        public IReadOnlyCollection<string> AstNodeIds { get; set; }

        [DataMember(Name = "argument")]
        public PickleStepArgument Argument { get; set; }

        public PickleStep()
        {
        }
        
        public PickleStep(string id, string text, PickleStepArgument argument, IEnumerable<string> astNodeIds)
        {
            Id = id;
            Text = text;
            Argument = argument;
            AstNodeIds = astNodeIds.ToReadOnlyCollection();
        }
    }
}
